pkg/strvals/parser.go

```diff
@@ -308,8 +308,11 @@ func typedVal(v []rune) interface{} {
 		return false
 	}
 
-	if iv, err := strconv.ParseInt(val, 10, 64); err == nil {
-		return iv
+	// If this value does not start with zero, try parsing it to an int
+	if len(val) != 0 && val[0] != 48 {
```

Comment:
48 is the ASCII value for `0`. Thanks for pointing this out, this prompted me to learn more about runes and I found out that a better implementation would be to use a rune literal `'0'` instead which better communicates the intent of the code.